core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/GetFromCacheHook.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.graph.hook;
+
+
+public interface GetFromCacheHook extends GraphHook {
+    String getSuffixCacheName();
```

Comment:
Why not implement a default method here to avoid the duplicate code in `NamedOperationResolver.java` & `NamedViewResolver.java`.

Reply:
because I need the cache instance to perform the query and I'd then need to add more methods like `getCache()` stubbing it all out.